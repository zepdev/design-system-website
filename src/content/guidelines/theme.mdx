---
label: Guidelines
title: Theme
---

# Theme

Themes are used to customize existing elements to fit the specific style of a strategic business unit or product. There is no base color for the Zeppelin-element-library, almost all elements require that a theme class is present. At the moment the Zeppelin Design System offers three main themes. Developers can customize all elements by adding the theme class to any html tag (`<main>`,`<body>`,`<div>`, ...) as long it wraps all elements you want to use.

```js
<main class="theme-zeppelin" id="app-root">
  ...
</main>
```

We currently offer three main themes;

```js
// Zeppelin Indigo Blue
<div class="theme-zeppelin" >...</div>

// CAT Yellow
<div class="theme-cat" >...</div>

// Rental Red
<div class="theme-rental" >...</div>
```
