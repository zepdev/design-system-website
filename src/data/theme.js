module.exports = {
  breakpoints: {
    xs: '36em' /* 576px */,
    s: '48em' /* 768px */,
    m: '62em' /* 992px */,
    l: '75em' /* 1200px */,
    xl: '105em' /* 1680px */,
    xxl: '120em' /* 1920px */,
  },
  typography: {
    typefaces: {
      roboto: "'Roboto', Arial, Helvetica, sans-serif",
      teko: "Teko', sans-serif",
    },
    basics: {
      fontSizeBase: {
        specification: 'font-size-base',
        px: '16px',
        rem: '1rem',
        weight: null,
      },
      lineHeightBase: {
        specification: 'line-height-base',
        px: '24px',
        rem: '1.5rem',
        weight: null,
      },
      fontWeightBase: {
        specification: 'font-weight-base',
        px: null,
        rem: null,
        weight: 400,
      },
    },
    headings: {
      normal: {
        'zep-typo--normal-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-5': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-6': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-7': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-8': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
      display: {
        'zep-typo--display-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
      special: {
        'zep-typo--special-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
    },
  },
  colors: {
    primary: {
      indigoBlue: {
        name: 'indigo-blue',
        hex: '#27166f',
        rgb: '39/22/111',
        hsb: null,
      },
      catYellow: {
        name: 'cat-yellow',
        hex: '#f0b823',
        rgb: '240/184/35',
        hsb: null,
      },
      rentalRed: {
        name: 'rental-red',
        hex: '#af0e0e',
        rgb: '175/14/14',
        hsb: null,
      },
    },
    gray: {
      grayDark: {
        name: 'gray-dark',
        hex: '#101111',
        rgb: '16/17/17',
        hsb: null,
      },
      grayMidDark: {
        name: 'gray-mid-dark',
        hex: '#373d3f',
        rgb: '55/61/63',
        hsb: null,
      },
      gray: { name: 'gray', hex: '#464a4c', rgb: '70/74/76', hsb: null },
      grayLight: {
        name: 'gray-light',
        hex: '#6f7c80',
        rgb: '111/124/128',
        hsb: null,
      },
      grayLighter: {
        name: 'gray-lighter',
        hex: '#eceeef',
        rgb: '236/238/239',
        hsb: null,
      },
      grayMid: {
        name: 'gray-mid',
        hex: '#f2f3f4',
        rgb: '242/243/244',
        hsb: null,
      },
      grayLightest: {
        name: 'gray-lightest',
        hex: '#f7f7f9',
        rgb: '247/247/249',
        hsb: null,
      },
      white: { name: 'white', hex: '#ffffff', rgb: '255/255/255', hsb: null },
    },
    support: {
      alert: { name: 'alert', hex: '#ff6562', rgb: '255/101/98', hsb: null },
      warning: {
        name: 'warning',
        hex: '#f0ad4e',
        rgb: '240,/73/78',
        hsb: null,
      },
      success: { name: 'success', hex: '#5cb85c', rgb: '92/184/92', hsb: null },
      info: { name: 'info', hex: '#5bc0de', rgb: '91/192/222', hsb: null },
    },
  },
  spacing: {
    xxs: 4,
    xs: 8,
    s: 12,
    m: 16,
    l: 24,
    xl: 32,
    xxl: 48,
  },
}
