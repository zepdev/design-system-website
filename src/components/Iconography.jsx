import React from 'react'
import PropTypes from 'prop-types'
import withStyles from 'react-jss'
import { iconography } from '../data/elements'

const styles = theme => ({
  root: {
    marginBottom: `${ theme.spacing.xxl.rem }rem`,
    display: 'flex',
    flexWrap: 'wrap',
  },
  container: {
    padding: `${ theme.spacing.m.rem }rem`,
    marginRight: `${ theme.spacing.m.rem }rem`,
  },
  img: {
    width: 32,
    height: 32,
  },
})

const Iconography = ({ classes }) => {
  return (
    <div className={classes.root}>
      {Object.keys(iconography).map((elem, idx) => (
        <div className={classes.container}>
          <img src={elem.src} alt={elem.title} className={classes.img} />
          <p className="zep-typo-normal-3">{elem.title}</p>
        </div>
      ))}
    </div>
  )
}

Iconography.propTypes = {
  classes: PropTypes.object.isRequired,
}

export default withStyles(styles)(Iconography)
