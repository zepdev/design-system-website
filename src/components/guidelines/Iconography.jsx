/* eslint-disable max-len */
import React, { useState, Fragment } from 'react'
import PropTypes from 'prop-types'
import clsx from 'clsx'
import withStyles from 'react-jss'
import ButtonBase from '../button/ButtonBase'
import DownloadIcon from '../icons/DownloadIcon'
import DownloadContainer from '../download/DownloadContainer'
import DownloadFile from '../download/DownloadFile'
import Tabs from '../tabs/Tabs'
import Tab from '../tabs/Tab'
import Divider from '../typography/Divider'
import Headline from '../typography/Headline'
import ElementUsage from '../elements/ElementUsage'
import { iconography } from '../../data/elements'

const styles = theme => ({
  tabs: {
    marginBottom: `${ theme.spacing.component.xxl.rem }rem`,
  },
  text: {
    marginBottom: `${ theme.spacing.component.l.rem }rem`,
  },
  root: {
    marginBottom: `${ theme.spacing.component.xxl.rem }rem`,
    display: 'flex',
    flexWrap: 'wrap',
  },
  container: {
    width: '50%',
    position: 'relative',
    background: theme.colors.gray.grayMid.hex,
    padding: 0,
    border: `1px solid ${ theme.colors.gray.white.hex }`,
    textAlign: 'center',
    '&:hover, &:focus': {
      zIndex: 1,
      '& $backdrop': {
        opacity: 0.9,
      },
      '& $backdropContent': {
        opacity: 1,
      },
    },
    '&:after': {
      content: "''",
      display: 'block',
      paddingBottom: '100%',
    },
  },
  [`@media (min-width: ${ theme.breakpoints.s })`]: {
    container: {
      width: '25%',
    },
  },
  [`@media (min-width: ${ theme.breakpoints.l })`]: {
    container: {
      width: '16.6%',
    },
  },
  [`@media (min-width: ${ theme.breakpoints.xl })`]: {
    container: {
      width: '12.5%',
    },
  },
  backdrop: {
    position: 'absolute',
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    backgroundColor: theme.colors.primary.indigoBlue.hex,
    opacity: 0,
    transition: 0.5,
    zIndex: 1,
  },
  backdropContent: {
    position: 'absolute',
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    opacity: 0,
    color: theme.colors.gray.white.hex,
  },
  icon: {
    color: theme.colors.gray.white.hex,
    zIndex: 2,
  },
  content: {
    position: 'absolute',
    height: '100%',
    width: '100%',
    padding: `${ theme.spacing.component.s.rem }rem`,
  },
  name: {
    textAlign: 'left',
    fontSize: 12,
    color: theme.colors.gray.black.hex,
  },
  img: {
    width: 32,
    height: 32,
    position: 'absolute',
    top: '50%',
    left: '50%',
    marginTop: -16,
    marginLeft: -16,
  },
})

const Iconography = ({ classes }) => {
  const [value, setValue] = useState(0)

  function handleChange(newValue) {
    setValue(newValue)
  }

  const zelVersion = '0.11.4' // we cannot call from package.json as it is always on next version which is not published to cdn.

  const sections = {
    action: {
      title: 'Action Icons',
    },
    ps: {
      title: 'Strategic Business Units Icons: Power Systems',
    },
    cat: {
      title: 'Strategic Business Units Icons: Zeppelin Cat',
    },
    systems: {
      title: 'Strategic Business Units Icons: Systems',
    },
    rental: {
      title: 'Strategic Business Units Icons: Rental',
    },
    navigation: {
      title: 'Navigation Icons',
    },
    indicator: {
      title: 'Indicator Icons',
    },
    language: {
      title: 'Language Icons',
    },
    social: {
      title: 'Social Icons',
    },
  }

  return (
    <>
      <Tabs value={value} onClick={handleChange} className={classes.tabs}>
        <Tab label="Overview" />
        <Tab label="Usage" />
      </Tabs>
      {value === 0 && (
        <>
          <Headline variant="md">Zeppelin Icons</Headline>
          <p className={clsx(classes.text, 'zep-typo--normal-body1')}>
            Icons can be downloaded as one large file containing all icons
            (Sketch, SVGs, or SVG Sprite) or icons may be downloaded
            individually from below.
          </p>
          <DownloadContainer>
            <DownloadFile
              title="Zeppelin Sketch Library"
              variant="sketch"
              href={`https://cdn-zel.zepdev.net/zel/${ zelVersion }/css/assets/dist/zds-library.sketch`}
              download=""
              ariaLabel="download"
            />
          </DownloadContainer>
          <DownloadContainer>
            <DownloadFile
              title="Zeppelin SVG Icons"
              href={`https://cdn-zel.zepdev.net/zel/${ zelVersion }/css/assets/dist/zepicons.zip`}
              download="zeppelin-iconography-library"
              ariaLabel="download"
            />
            <DownloadFile
              title="Zeppelin SVG Sprite"
              href={`https://cdn-zel.zepdev.net/zel/${ zelVersion }/css/assets/dist/zepsprite.zip`}
              download=""
              ariaLabel="download"
            />
          </DownloadContainer>
          <Divider />
          {Object.keys(sections).map(section => (
            <Fragment key={`section${ section }`}>
              <Headline variant="md">{sections[section].title}</Headline>
              <div className={classes.root}>
                {iconography[section].map(elem => (
                  <ButtonBase
                    className={classes.container}
                    key={`icon${ elem }`}
                    target="_blank"
                    href={require(`zeppelin-element-library/bundle/assets/icons/SVG/zepicons-${ elem }.svg`)}
                    download={`zepicons-${ elem }.svg`}
                    aria-label={elem}
                  >
                    <div className={classes.content}>
                      <span className={classes.backdrop} />
                      <span className={classes.backdropContent}>
                        <DownloadIcon
                          className={classes.icon}
                          ariaLabel="download"
                        />
                      </span>
                      <p
                        className={clsx(
                          classes.name,
                          'zep-typo--normal-caption'
                        )}
                      >{`zepicons-${ elem }`}</p>
                      <img
                        src={require(`zeppelin-element-library/bundle/assets/icons/SVG/zepicons-${ elem }.svg`)}
                        alt={elem}
                        className={classes.img}
                      />
                    </div>
                  </ButtonBase>
                ))}
              </div>
            </Fragment>
          ))}
        </>
      )}
      {value === 1 && <ElementUsage element="iconography" />}
    </>
  )
}

Iconography.propTypes = {
  classes: PropTypes.object.isRequired,
}

export default withStyles(styles)(Iconography)
