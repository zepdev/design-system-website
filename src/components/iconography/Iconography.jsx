import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import withStyles from 'react-jss'
import ButtonBase from '../button/ButtonBase'
import DownloadIcon from '../icons/DownloadIcon'
import { iconography } from '../../data/elements'

const styles = theme => ({
  root: {
    marginBottom: `${ theme.spacing.xxl.rem }rem`,
    display: 'flex',
    flexWrap: 'wrap',
  },
  container: {
    width: '50%',
    position: 'relative',
    background: theme.colors.gray.grayMid.hex,
    padding: 0,
    border: `1px solid ${ theme.colors.gray.white.hex }`,
    textAlign: 'center',
    '&:hover, &:focus': {
      zIndex: 1,
      '& $backdrop': {
        opacity: 0.9,
      },
      '& $backdropContent': {
        opacity: 1,
      },
    },
    '&:after': {
      content: "''",
      display: 'block',
      paddingBottom: '100%',
    },
  },
  [`@media (min-width: ${ theme.breakpoints.s })`]: {
    container: {
      width: '25%',
    },
  },
  [`@media (min-width: ${ theme.breakpoints.l })`]: {
    container: {
      width: '16.6%',
    },
  },
  [`@media (min-width: ${ theme.breakpoints.xl })`]: {
    container: {
      width: '12.5%',
    },
  },
  [`@media (min-width: ${ theme.breakpoints.xxl })`]: {
    container: {
      width: '8.3%',
    },
  },
  backdrop: {
    position: 'absolute',
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    backgroundColor: theme.colors.primary.indigoBlue.hex,
    opacity: 0,
    transition: 0.5,
    zIndex: 1,
  },
  backdropContent: {
    position: 'absolute',
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    opacity: 0,
    color: theme.colors.gray.white.hex,
  },
  icon: {
    color: theme.colors.gray.white.hex,
    zIndex: 2,
  },
  content: {
    position: 'absolute',
    height: '100%',
    width: '100%',
    padding: `${ theme.spacing.s.rem }rem`,
  },
  text: {
    textAlign: 'left',
    fontSize: 12,
  },
  img: {
    width: 32,
    height: 32,
    position: 'absolute',
    top: '50%',
    left: '50%',
    marginTop: -16,
    marginLeft: -16,
  },
})

const Iconography = ({ section, classes }) => {
  const icon = 'IconsMiscActionBell'

  return (
    <div className={classes.root}>
      {iconography[section].map(elem => (
        <ButtonBase
          className={classes.container}
          key={`icon${ elem }`}
          target="_blank"
          href={require(`../../assets/icons/${ icon }.svg`)}
          download={`zepicons-${ elem }.svg`}
          aria-label={elem}
        >
          <div className={classes.content}>
            <span className={classes.backdrop} />
            <span className={classes.backdropContent}>
              <DownloadIcon className={classes.icon} ariaLabel="download" />
            </span>
            <p className={classnames(classes.text, 'zep-typo--normal-1')}>{`zepicons-${ elem }`}</p>
            <img
              src={require(`../../assets/icons/${ icon }.svg`)}
              alt={elem}
              className={classes.img}
            />
          </div>
        </ButtonBase>
      ))}
    </div>
  )
}

Iconography.propTypes = {
  classes: PropTypes.object.isRequired,
  section: PropTypes.string.isRequired,
}

export default withStyles(styles)(Iconography)
