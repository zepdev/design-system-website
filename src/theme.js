import { createMuiTheme } from '@material-ui/core/styles'
import guidelines from '@zlab-de/zel-react/guidelines.json'

const theme = createMuiTheme({
  palette: {
    type: 'light',
    primary: {
      light: guidelines.theme.indigo.secondary,
      main: guidelines.theme.indigo.primary,
      dark: guidelines.theme.indigo.primary,
    },
    error: { main: guidelines.color.support.alert },
  },
  status: {
    gray: '#eceeef',
  },
  logo: guidelines.logo,
  blue: guidelines.theme.blue,
  indigo: guidelines.theme.indigo,
  red: guidelines.theme.red,
  yellow: guidelines.theme.yellow,
  guidelines: guidelines,
  space: guidelines.spacing.component,
  color: guidelines.color,
})

export default theme
