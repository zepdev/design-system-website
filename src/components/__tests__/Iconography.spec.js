import React from 'react'
import { render } from 'react-testing-library'
import { ThemeProvider } from 'react-jss'
import theme from '../../data/theme'
import Iconography from '../Iconography'

describe('Iconography', () => {
  it('renders correctly', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Iconography />
      </ThemeProvider>
    )
    expect(component).toMatchSnapshot()
  })
})
