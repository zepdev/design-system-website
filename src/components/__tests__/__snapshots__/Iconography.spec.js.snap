// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Iconography renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Iconography-root-0-1-1"
      >
        <div
          class="Iconography-container-0-1-2"
        >
          <img
            alt="zep-logo"
            class="Iconography-img-0-1-3"
            src="https://s3.eu-central-1.amazonaws.com/com.zeppelin.zds.assets/zel/zeppelin_icon.png"
          />
          <p
            class="zep-typo-normal-3"
          >
            zep-logo
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Iconography-root-0-1-1"
    >
      <div
        class="Iconography-container-0-1-2"
      >
        <img
          alt="zep-logo"
          class="Iconography-img-0-1-3"
          src="https://s3.eu-central-1.amazonaws.com/com.zeppelin.zds.assets/zel/zeppelin_icon.png"
        />
        <p
          class="zep-typo-normal-3"
        >
          zep-logo
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
